### PR TITLE
BSP-2947 Improve JS debugging on latest styleguide

### PR DESCRIPTION
**Pull request by mlikouris**
In the gulp watch process, if environment variable JS_MIN doesn't equal true then copy All.js to All.min.js

### CODE REVIEW

File: js/task/watch.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
+/* eslint-disable no-unused-vars */
```
Comment (jpencola) [inline js/task/watch.js]: Escape the linter only on the LOC where it's needed

File: js/task/watch.js [hunk 2]
```diff
@@ -1,4 +1,10 @@
...
 const logger = require('../logger')
+const plugins = require('gulp-load-plugins')()
+const path = require('path')
```
Comment (jpencola) [inline js/task/watch.js]: Reorder the imports alphabetically

**Review by jpencola — CHANGES_REQUESTED** [review]

File: js/task/watch.js (hunk 2 above)
Comment (mlikouris) [inline js/task/watch.js]: gulp-load-plugins, comes before path. ;)

**Review by jpencola — APPROVED** [review]

File: js/task/watch.js [hunk 3]
```diff
@@ -28,6 +33,16 @@ module.exports = (styleguide, gulp) => {
   styleguide.watch.js = () => {
     gulp.watch('styleguide/**/*.js', { cwd: styleguide.path.root() }, deps(styleguide.task.js()))
       .on('change', onChange)
+    // waiting for change in the minified file before copying unminified to minified
+    if (process.env.JS_MIN !== 'true') {
+      gulp.watch('styleguide/All.min.js', { cwd: styleguide.path.build() })
+        .on('change', function (event) {
+          logger.info(`COPY .js >> .min.js: ${event.path}`)
```
Comment (jpencola) [inline js/task/watch.js]: I think this log message should be removed because it makes the watcher log too verbose.

**Review by jpencola — CHANGES_REQUESTED** [review]

**Review by jpencola — APPROVED** [review]
💯

**Pull request closed without merging**